plugins/scaffolder-backend/src/ScaffolderPlugin.ts

```diff
@@ -91,20 +102,39 @@ export const scaffolderPlugin = createBackendPlugin({
         permissions,
       }) {
         const log = loggerToWinstonLogger(logger);
+        const integrations = ScmIntegrations.fromConfig(config);
 
         const actions = [
+          // actions provided from other modules
           ...addedActions,
-          ...createBuiltinActions({
-            integrations: ScmIntegrations.fromConfig(config),
-            catalogClient,
+
+          // built-in actions for the scaffolder
+          createFetchPlainAction({
+            reader,
+            integrations,
+          }),
+          createFetchPlainFileAction({
+            reader,
+            integrations,
+          }),
+          createFetchTemplateAction({
+            integrations,
             reader,
-            config,
             additionalTemplateFilters,
             additionalTemplateGlobals,
           }),
+          createDebugLogAction(),
+          createWaitAction(),
+          // todo(blam): maybe these should be a -catalog module?
+          createCatalogRegisterAction({ catalogClient, integrations }),
+          createFetchCatalogEntityAction({ catalogClient }),
+          createCatalogWriteAction(),
```

Comment:
Feels like we'd only want to do that from a PoV of keeping the architecture clean. They all have minimal impact on the dependency tree and fairly minimal impact on security as well, although that's perhaps where I'd see the biggest reason to keep them separate. In the end I think it makes sense to have them built-in though given that the scaffolder doesn't function without the catalog anyway, which doesn't apply to the catalog template kind module. If really needed we could add a config flag for disabling these too I guess